mlptrain/sampling/metadynamics.py

```diff
@@ -0,0 +1,165 @@
+
+            cvs (Sequence): Sequence of PLUMED collective variable objects
+        """
+        self.bias:     mlptrain.PlumedBias = PlumedBias(cvs)
+        self.temp:     Optional[float] = temp                     # K
+
+    def run_metadynamics(self,
+                         start_config: 'mlptrain.Configuration',
+                         mlp: 'mlptrain.potentials._base.MLPotential',
+                         temp: float,
+                         interval: int,
+                         dt: float,
+                         pace: int,
+                         width: float,
+                         height: float,
+                         biasfactor: float,
+                         n_runs: int = 1,
+                         n_walkers: int = 1,
+                         save_sep: bool = False,
+                         **kwargs) -> None:
+        """
+        Perform multiple well-tempered metadynamics runs in parallel, generate
+        .xyz files containing trajectories of the runs, generate PLUMED files
+        containing deposited gaussians and trajectories in terms of the CVs.
+
+        -----------------------------------------------------------------------
+        Arguments:
+
+            start_config: Configuration from which the simulation is started
+
+            mlp: Machine learnt potential
+
+            temp (float): Temperature in K to initialise velocities and to run
+                          NVT MD. Must be positive
+
+            interval (int): Interval between saving the geometry
+
+            dt (float): Time-step in fs
+
+            pace (int): τ_G/dt, interval at which a new gaussian is placed
+
+            width (float): σ, standard deviation (parameter describing the
+                           width) of the placed gaussian
+
+            height (float): ω, initial height of placed gaussians
+
+            biasfactor (float): γ, describes how quickly gaussians shrink,
+                                larger values make gaussians to be placed
+                                less sensitive to the bias potential
+
+            n_runs (int): Number of times to run metadynamics
+
+            n_walkers (int): Number of walkers to use in each simulation
+
+            save_sep (bool): If True saves trajectories of
+                             each window separately
+
+        -------------------
+        Keyword Arguments:
+
+            {fs, ps, ns}: Simulation time in some units
+        """
+        if temp <= 0:
+            raise ValueError('Temperature must be positive and non-zero for '
+                             'well-tempered metadynamics')
+
+        start_metadynamics = time.perf_counter()
+
+        self.temp = temp
+        self.bias.set_metad_params(width=width,
+                                   pace=pace,
+                                   height=height,
+                                   biasfactor=biasfactor)
+
+        metad_processes, metad_trajs = ([] for _ in range(2))
```

Comment:
I generally don't use this for small numbers of lists. It's longer and more difficult to read than simply: = [], []